src/misc.cpp

```diff
@@ -221,6 +225,19 @@ const std::string compiler_info() {
   return compiler;
 }
 
+/// large_page_alloc() tries to allocate aligned large page memory on systems
+/// where available, if no large page memory is available, small page memory
+/// is allocated instead.
+void* large_page_alloc(size_t alignment, size_t size) {
+  auto addr = aligned_alloc(alignment, size);
```

Comment:
This does not guarantee page-alignnment, unless size = 4096 or larger. 2M is preferred for transparent huge pages on x86. This said, for big alloc sizes (like >= 16 MB) Linux/glibc usually gives 4kB page-aligned memory.

Reply:
Do we need to manually align to 2MB? I always get correctly aligned huge pages when I just set the advise strategy.

Reply:
I only get 4-kB aligned memory even with gigabytes of alloc sizes. However, this actually only affects the very beginning and the end of the buffer. If the buffer is not 2-MB-aligned, then the ends will use 4-kB pages after madvise(), but the buffer interior will still use 2-MB pages.

There is a small difference for me in "cat /proc/meminfo | grep AnonHugePages" depending on whether alignment is used or not.

Reply:
(Edited a typo: I meant unless alignment >= 4096)

Reply:
I see, so in my version there are a number of 4k pages at the start before the huge page memory starts.

Reply:
At the start and at the end. But I would have thought the perf difference is close to negligible, and not 3% like @MichaelB7 reported.

Reply:
The difference is negligible on my system.